pylint/lint/pylinter.py

```diff
@@ -1005,6 +1014,33 @@ def initialize(self):
             if not msg.may_be_emitted():
                 self._msgs_state[msg.msgid] = False
 
+    @staticmethod
+    def _discover_files(files_or_modules: Sequence[str]) -> Iterator[str]:
```

Comment:
Shouldn't `yields` be typed with `Generator`? That's what I always do.

Reply:
It it is just yielding it can be also Iterator - see https://docs.python.org/3/library/typing.html#typing.Generator. But if you insist to have here Generator type, I am not against it - please let me know.

Reply:
TIL! Thanks 😄 

Fine with keeping like this